sharktank/sharktank/ops/sharded_impls.py

```diff
@@ -1075,6 +1075,45 @@ def module_register_buffer_sharded(
     setattr(module, name, tensor)
 
 
+@ones.override()
+def ones_replicated(
+    *args,
+    devices: Sequence[int] | None = None,
+    **kwargs,
+):
+    if devices is None:
+        return NotImplemented
+
+    # Do not use `tranfer_to_logical_device` here.
+    # Adding the transfer op would prevent the ones result from being fused.
+    shards = [ones(*args, **kwargs) for _ in devices]
+    return ReplicatedTensor(ts=shards, devices=tuple(devices))
+
+
+@ones_like.override(AllOfType(ReplicatedTensor, SplitPrimitiveTensor))
+def ones_like_replicated(
+    tensor: ReplicatedTensor | SplitPrimitiveTensor,
+    *,
+    dtype: torch.dtype | None,
+    layout: torch.layout | None,
+    device: torch.device | None,
+    requires_grad: bool,
+    memory_format: torch.memory_format,
```

Comment:
Don't include features that are meaningless to compiled models (e.g. memory_format)